pkg/development/devcontext.go

```diff
@@ -155,17 +156,47 @@ func loadTuples(ctx context.Context, tuples []*core.RelationTuple, rwt datastore
 	return devErrors, err
 }
 
-func loadNamespaces(
+func loadCompiled(
 	ctx context.Context,
-	namespaces []*core.NamespaceDefinition,
+	compiled *compiler.CompiledSchema,
 	rwt datastore.ReadWriteTransaction,
 ) ([]*devinterface.DeveloperError, error) {
-	errors := make([]*devinterface.DeveloperError, 0, len(namespaces))
+	errors := make([]*devinterface.DeveloperError, 0, len(compiled.OrderedDefinitions))
 	resolver := namespace.ResolverForPredefinedDefinitions(namespace.PredefinedElements{
-		Namespaces: namespaces,
+		Namespaces: compiled.ObjectDefinitions,
+		Caveats:    compiled.CaveatDefinitions,
 	})
 
-	for _, nsDef := range namespaces {
+	for _, caveatDef := range compiled.CaveatDefinitions {
+		cverr := namespace.ValidateCaveatDefinition(caveatDef)
```

Comment:
> Or is it that this will only be used by memdb in spicedb via WASM and is all running in the user's browser?

Exactly, and I just followed the namespace pattern below. If you like, I can validate the caveats before hand, but its all under a memdb transaction, so I'm unsure if it really matters

Reply:
I guess it may not mean much for memdb in the browser